bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,263 @@
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.vektiva.internal.config.VektivaSmarwiConfiguration;
+import org.openhab.binding.vektiva.internal.net.VektivaSmarwiSocket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VektivaSmarwiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class VektivaSmarwiHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VektivaSmarwiHandler.class);
+
+    private VektivaSmarwiConfiguration config = new VektivaSmarwiConfiguration();
+
+    private final HttpClient httpClient;
+
+    private final WebSocketClient webSocketClient;
+
+    private @Nullable Session session;
+
+    private @Nullable ScheduledFuture<?> future = null;
+
+    private int lastPosition = -1;
+
+    public VektivaSmarwiHandler(Thing thing, HttpClient httpClient, WebSocketClient webSocketClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.webSocketClient = webSocketClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            checkStatus();
+            return;
+        }
+
+        if (CHANNEL_CONTROL.equals(channelUID.getId())) {
+            logger.debug("Received command: {}", command);
+            String cmd = getSmarwiCommand(command);
+            if (COMMAND_OPEN.equals(cmd) || COMMAND_CLOSE.equals(cmd) || COMMAND_STOP.equals(cmd)) {
+                if (RESPONSE_OK.equals(sendCommand(cmd))) {
+                    lastPosition = COMMAND_OPEN.equals(cmd) ? 0 : 100;
+                }
+            }
+            if (command instanceof PercentType) {
+                if (RESPONSE_OK.equals(sendCommand(COMMAND_OPEN + "/" + cmd))) {
+                    lastPosition = Integer.parseInt(cmd);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (future != null && !(future.isCancelled() || future.isDone())) {
+            future.cancel(true);
+        }
+        closeSession();
+    }
+
+    private void closeSession() {
+        if (session != null && session.isOpen()) {
+            session.close();
+        }
+    }
+
+    private String getSmarwiCommand(Command command) {
+        switch (command.toString()) {
```

Comment:
Comparing them by converting them to Strings is not that great :-) underlying they are just enums.

Reply:
If you convert it to an if its a little more efficient and less hacky

Reply:
converted to if